src/librustc_mir/build/block.rs

```diff
@@ -30,28 +36,57 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                     let exit_block = this.cfg.start_new_block();
                     let block_exit = this.in_breakable_scope(
                         None, exit_block, destination.clone(), |this| {
-                            this.ast_block_stmts(destination, block, span, stmts, expr,
-                                                 safety_mode)
+                            // We can have multiple assignments to the
+                            // destination, so we cannot schedule drop after
+                            // each assignment. We have to rely on drop
+                            // elaboration for this.
```

Comment:
I see this comment in a few places but I am confused by it, hopefully answering these can help:
* "we can have multiple assignments" where? maybe talk about *what* can contain them?
* "so we cannot" doesn't explain why we're scheduling a drop *here*. maybe saying "so we're doing X instead" would be clearer
* and what's the "this" we rely on drop elaboration for? removing noop drops? making drops conditional? how does it relate to everything above?